glossary.md

```diff
@@ -0,0 +1,241 @@
+the execution engine must include in the block and execute.
+
+> **TODO** The transaction field must be specified.
+
+Payload attributes were historically called "L2 block inputs" and you might
+still hear some people using this term.
+
+## L1 Attributes Transaction
+[l1-attributes-tx]: /glossary.md#l1-attributes-transaction
+
+A transaction with an Optimistic-Ethereum-specific transaction type, that is
+used to register the L1 block attributes (number, timestamp, ...) on L2.
+
+The L1 attributes for a given L1 block can be read on L2 from the [L1 Attributes
+Precompiled Contract][l1-attr-precompile].
+
+cf. [L1 attributes transaction format](/rollup-node.md#payload-transaction-format)
+(in the section on [payload attributes])
+
+> **TODO** We might want to move this the format spec to the execution engine.
+
+## L1 Attributes Precompiled Contract
+[l1-attr-precompile]: /glossary.md#l1-attributes-precompiled-contract
+
+A [precompiled contract][precompile] on L2 that can be used to retrieve the L1
+block attributes of L1 blocks with a given block number or a given block hash.
+
+cf. [L1 Attributes Precompiled Contract Specification](TODO)
+
+> **TODO LINK** L1 attributes precompiled contract spec
+
+## Deposits
+[deposits]: /glossary.md#deposits
+
+A deposit is an L2 transaction that has been submitted on L1, via a transaction
+sent to the [L2 deposit feed contract][deposit-feed].
+
+While deposits are notably (but not only) used to "deposit" (bridge) ETH and
+tokens to L2, the word *deposit* should be understood as "a transaction
+*deposited* to L2".
+
+Deposits are one kind of [L2 derivation input][deriv-input].
+
+## L2 Deposit Feed Contract
+[deposit-feed]: /glossary.md#L2-deposit-feed-contract
+
+A contract to which [EOAs][EOA] and contracts may send [deposits]. The deposits
+are emitted as log records (in Solidity, these are called *events*) for
+consumption by [rollup nodes][rollup node].
+
+Advanced note: the deposits are not stored in calldata because they can be send
+by contracts, in which case the calldata is part of the execution, but its value
+is not captured in one of the [Merkle roots][Merkle root] included in the L1
+block.
+
+cf. [L2 Deposit Feed Contract Specification](TODO)
+
+> **TODO LINK** deposit feed contract specification
+
+## Block Gossip
+[block gossip]: /glossary.md#block-gossip
+
+Block gossip is a [rollup node] component responsible for transmitting the L2
+blocks derived by the rollup node over a peer-to-peer network.
+
+> Note that block gossip is not useful in the current deposit-only model, but
+> will become useful in later stages of the specification process.
+
+--------------------------------------------------------------------------------
+
+# Execution Engine Concepts
+
+## Execution Engine
+[execution engine]: /glossary.md#execution-engine
+
+The execution engine is responsible for executing transactions in blocks and
+computing the resulting state roots, log roots and block hash.
```

Comment:
There is no such thing as a "log root". The logs are part of the receipt (encoded in RLP etc.), but not merkleized or hashed individually. The receipts form a MPT. They are combined into a bloom filter though.

Reply:
You're right, "receipts root" is really what is meant here.